### PR TITLE
Add test for OTel Collector start

### PR DESCRIPTION
**Pull request by andrzej-stencel**
## What does this PR do?

Adds a test that verifies that the OTel Collector embedded in the Elastic Agent can be started successfully with all the components included in it.

## Why is it important?

We want to make sure all the components we include in the OTel distro can be actually run - e.g. there are no runtime crashes, port conflicts, etc.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- ~~[ ] I have made corresponding changes to the documentation~~
- ~~[ ] I have made corresponding change to the default configuration files~~
- [x] I have added tests that prove my fix is effective or that my feature works
- ~~[ ] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/elastic-agent#changelog)~~
- ~~[ ] I have added an integration test or an E2E test~~

## How to test this PR locally

```shell
cd internal/pkg/otel
go test -v .
```

## Related issues

- Relates https://github.com/elastic/elastic-agent/issues/4723

### CODE REVIEW

File: internal/pkg/otel/run_test.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+		},
+		{
+			configFile:           "nonexistent-component.yml",
+			expectedErrorMessage: `error decoding 'extensions': unknown type: "zpages"`,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.configFile, func(t *testing.T) {
+			configFiles := []string{"file:" + filepath.Join(".", "testdata", tc.configFile)}
```
Comment (blakerouse) [inline internal/pkg/otel/run_test.go:36]: Surprised that doesn't need to be `file://`

**Review by blakerouse — APPROVED** [review]
Looks good.

File: internal/pkg/otel/run_test.go (hunk 1 above)
Comment (andrzej-stencel) [reply internal/pkg/otel/run_test.go:36]: Yeah, "file" is the default when there's no scheme provided.

- https://opentelemetry.io/docs/collector/configuration/#location
- https://github.com/open-telemetry/opentelemetry-collector/blob/v0.100.0/confmap/resolver.go#L82-L83
- https://github.com/open-telemetry/opentelemetry-collector/blob/v0.100.0/confmap/resolver.go#L131-L132

**Comment by andrzej-stencel** [comment]
I need to fix the test for Windows, converting to draft PR until I fix this.

File: internal/pkg/otel/run_test.go (hunk 1 above)
Comment (andrzej-stencel) [reply internal/pkg/otel/run_test.go:36]: Oh you mean the `//` in `file://`. Yeah that's how it works.

**Comment by andrzej-stencel** [comment]
The test is fixed now.

File: internal/pkg/otel/run_test.go (hunk 1 above)
Comment (blakerouse) [reply internal/pkg/otel/run_test.go]: What do you mean that is how it works? It works with `file://` or `file:`? I was just stating that I am surprised if doesn't require the `//` before the actual path.

File: internal/pkg/otel/testdata/all-components.windows.yml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+exporters:
+  file:
+    path: ${env:TMP}\file-exporter-output.json
```
Comment (blakerouse) [inline internal/pkg/otel/testdata/all-components.windows.yml:3]: seems that windows supports a lot less? is this from upstream or just how we have enabled it?

File: internal/pkg/otel/run_test.go (hunk 1 above)
Comment (andrzej-stencel) [reply internal/pkg/otel/run_test.go]: Yeah, by "this is how it works" I meant that this is how it's implemented in the file provider: https://github.com/open-telemetry/opentelemetry-collector/blob/v0.100.0/confmap/provider/fileprovider/provider.go#L42-L73.

Whether it's compliant with the https://en.wikipedia.org/wiki/Uniform_Resource_Identifier#Syntax or https://en.wikipedia.org/wiki/File_URI_scheme or https://datatracker.ietf.org/doc/html/rfc3986 is probably another question. The file provider definitely allows relative paths like `file:./local-file.txt`, which I'm not sure the RFC does.

File: internal/pkg/otel/run_test.go [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+// getConfigFiles returns a collection of config file paths for the collector to use.
+// In the simplest scenario, the collection will contains only one path.
+// In case there is an operating system-specific override file found, it will be added to the collection.
+// E.g. if the input file name is `all-components.yml` and a file named `all-components.windows.yml` exists,
+// the config path collection will have two elements on Windows, and only one element on other OSes.
+// Use `darwin` for MacOS, `linux` for Linux and `windows` for Windows.
+func getConfigFiles(configFileName string) []string {
+	// Add base file to the collection.
+	baseFilePath := filepath.Join(".", "testdata", configFileName)
```
Comment (pchila) [inline internal/pkg/otel/run_test.go:69]: Is the `.` initial token of the path required by the `file:` location? Aside from the `//` that was already discussed in Blake's comments I wonder if omitting the current directory would still work. something like
```suggestion
	baseFilePath := filepath.Join("testdata", configFileName)
```

**Review by pchila — APPROVED** [review]
Looks good.
Just a little curiosity about the exact semantics of `file:` locator but that is obviously non-blocking

File: internal/pkg/otel/run_test.go (hunk 3 above)
Comment (andrzej-stencel) [reply internal/pkg/otel/run_test.go:69]: Yes, this should work equally well.

**Pull request merged**